### PR TITLE
Migrate jobs to --skew and --extract=JENKINS_PUBLISHED_VERSION

### PR DESCRIPTION
**Pull request by fejta**
ref https://github.com/kubernetes/test-infra/issues/2829

Using a `SKEW_KUBECTL=y` for the time being, will migrate that one in the future.

### CODE REVIEW

File: jobs/ci-kubernetes-e2e-gce-1-6-master-upgrade-cluster.env [hunk 1]
```diff
@@ -5,8 +5,8 @@ E2E_OPT=--check_version_skew=false
 E2E_UPGRADE_TEST=true
 STORAGE_MEDIA_TYPE=application/vnd.kubernetes.protobuf
 GINKGO_UPGRADE_TEST_ARGS=--ginkgo.focus=\[Feature:ClusterUpgrade\] --upgrade-target=ci/latest
-JENKINS_PUBLISHED_SKEW_VERSION=ci/latest
-JENKINS_PUBLISHED_VERSION=ci/latest-1.6
 KUBE_NODE_OS_DISTRIBUTION=debian
 PROJECT=gce-up-c1-3-g1-4-up-clu
 
+
+SKEW_KUBECTL=y
```
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-gce-1-6-master-upgrade-cluster.env]: new line

File: jobs/config.json [hunk 2]
```diff
@@ -195,7 +205,9 @@
     "args": [
       "--env-file=platforms/gce.env", 
       "--env-file=jobs/ci-kubernetes-e2e-gce-1.5-1.6-cvm-kubectl-skew.env", 
-      "--timeout=120m"
+      "--timeout=120m", 
+      "--extract=ci/latest-1.6", 
+      "--extract=ci/latest-1.5"
```
Comment (krzyzacy) [inline jobs/config.json]: it does not look right..

**Comment by krzyzacy** [comment]
Looks super cool overall, I'd like to refactor and share the env files across all releases :-)

File: jobs/config.json (hunk 2 above)
Comment (fejta) [inline jobs/config.json]: It is right, the old values were:

```
JENKINS_PUBLISHED_SKEW_VERSION=ci/latest-1.6		
JENKINS_PUBLISHED_VERSION=ci/latest-1.5
```

And if you look at the old `e2e-runner.sh logic` you'll notice that the `JENKINS_PUBLISHED_SKEW_VERSION --extract` gets added first, same as here.

File: jobs/ci-kubernetes-e2e-gce-1-6-master-upgrade-cluster.env (hunk 1 above)
Comment (fejta) [inline jobs/ci-kubernetes-e2e-gce-1-6-master-upgrade-cluster.env]: Done

**Comment by fejta** [comment]
Makes sense, although I'd like to leave that out of this PR :stuck_out_tongue:

File: jobs/config.json (hunk 2 above)
Comment (krzyzacy) [inline jobs/config.json]: hummm we only want to keep one I assume?

Comment (fejta) [inline jobs/config.json]: No, this is an upgrade scenario. We need to download extract two versions.

**Comment by enisoc** [comment]
@fejta I think this has caused all my `release-1.6` jobs to start testing against `master` builds.

For example:

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-release-1-6/42

**Comment by krzyzacy** [comment]
@enisoc probably need https://github.com/kubernetes/test-infra/pull/2974, I'll merge that.

**Pull request merged**